src/views/Nft/market/hooks/useCollectionNfts.tsx

```diff
@@ -0,0 +1,167 @@
+  const subgraphRes = await getNftsMarketData(
+    whereClause,
+    REQUEST_SIZE,
+    settings.field,
+    settings.direction,
+    page * REQUEST_SIZE,
+  )
+  const apiRequestPromises = subgraphRes.map((marketNft) => getNftApi(address, marketNft.tokenId))
+  const apiResponses = await Promise.all(apiRequestPromises)
+  const newNfts: NftToken[] = apiResponses.reduce((acc, apiNft) => {
+    if (apiNft) {
+      acc.push({
+        ...apiNft,
+        collectionAddress: address,
+        collectionName: apiNft.collection.name,
+        marketData: subgraphRes.find((marketNft) => marketNft.tokenId === apiNft.tokenId),
+      })
+    }
+    return acc
+  }, [] as NftToken[])
+  return newNfts
+}
+
+const fetchAllNfts = async (address: string, settings: ItemListingSettings, page: number): Promise<NftToken[]> => {
+  const tokenIdsFromFilter = await fetchTokenIdsFromFilter(address, settings)
+
+  let collectionNftsResponse: ApiResponseCollectionTokens = null
+  let tokenIds = null
+
+  if (tokenIdsFromFilter) {
+    tokenIds = tokenIdsFromFilter
+  } else {
+    collectionNftsResponse = await getNftsFromCollectionApi(address, REQUEST_SIZE, page)
+    if (collectionNftsResponse?.data) {
+      tokenIds = Object.values(collectionNftsResponse.data).map((nft) => nft.tokenId)
+    }
+  }
+
+  if (tokenIds) {
+    const nftsMarket = await getMarketDataForTokenIds(address, tokenIds)
+
+    const responsesPromises = tokenIds.map(async (id) => {
+      const apiMetadata: ApiSingleTokenData = collectionNftsResponse
+        ? collectionNftsResponse.data[id]
+        : await getNftApi(address, id)
+      const marketData = nftsMarket.find((nft) => nft.tokenId === id)
+
+      return {
+        tokenId: id,
+        name: apiMetadata.name,
+        description: apiMetadata.description,
+        collectionName: apiMetadata.collection.name,
+        collectionAddress: address,
+        image: apiMetadata.image,
+        attributes: apiMetadata.attributes,
+        marketData,
+      }
+    })
+
+    const newNfts: NftToken[] = await Promise.all(responsesPromises)
+    return newNfts
+  }
+  return []
+}
+
+export const useCollectionNfts = (collectionAddress: string) => {
+  const { field, direction } = useGetNftOrdering(collectionAddress)
+  const showOnlyNftsOnSale = useGetNftShowOnlyOnSale(collectionAddress)
+  const nftFilters = useGetNftFilters(collectionAddress)
+  const [itemListingSettings, setItemListingSettings] = useState<ItemListingSettings>({
+    field,
+    direction,
+    showOnlyNftsOnSale,
+    nftFilters,
+  })
+
+  const itemListingSettingsJson = JSON.stringify(itemListingSettings)
+  const filtersJson = JSON.stringify(nftFilters)
+
+  useEffect(() => {
+    setItemListingSettings(() => ({
+      field,
+      direction,
+      showOnlyNftsOnSale,
+      nftFilters: JSON.parse(filtersJson),
+    }))
+  }, [field, direction, showOnlyNftsOnSale, filtersJson])
+
+  const {
+    data: nfts,
+    isValidating,
```

Comment:
Showing loading when validating is not a good idea

Reply:
Fixed